draft-ietf-quic-tls.md

```diff
@@ -2220,6 +2221,81 @@ is, `2l = 2^11`). This simplification is based on the packet containing all of
 the associated data and ciphertext. This results in a negligible 1 to 3 block
 overestimation of the number of operations.
 
+## Analysis AEAD_AES_128_GCM and AEAD_AES_256_GCM Usage Limits {#gcm-bounds}
+
+{{?GCM-MU}} specify concrete bounds for AEAD_AES_128_GCM and AEAD_AES_256_GCM as
+used in TLS 1.3 and QUIC. This section documents this analysis using several
+simplifying assumptions:
+
+- The number of ciphertext blocks an attacker uses in forgery attempts is bounded
+by v * l, the number of forgery attempts and the size of each packet (in blocks).
+
+- The amount of offline work done by an attacker does not dominate other factors
+in the analysis.
+
+The bounds in {{?GCM-MU}} are tighter and more complete than those used in
+{{AEBounds}}, which allows for larger limits than those described in {{?TLS13}}.
+
+### Confidentiality Limit
+
+For confidentiality, Equation (1) in {{?GCM-MU}} establishes that an attacker gains
+a distinguishing advantage between a real and random AEAD algorithm of no more than:
+
+~~~
+((q + v) * l)^2 / 2^128
+~~~
+
+For a target advantage of 2^-60, which matches that used by {{!TLS13}}, this
+results in the relation:
+
+~~~
+(q + v) <= 2^28
+~~~
+
+Assuming `v = q`, endpoints cannot protect more than 2^27 packets in a single
+connection without causing an attacker to gain an larger advantage than the
+target of 2^-60.
+
+### Integrity Limit
+
+For integrity, Theorem (4.3) in {{?GCM-MU}} establishes that an attacker gains
+an advantage in successfully forging a packet of no more than:
+
+~~~
+(1 / 2^(8 * n)) + ((2 * v) / 2^2*n) + ((2 * o * v) / 2^(k + n))
+        + (n * (v + (v * l)) / 2^k)
+~~~
+
+The goal is to limit this advantage to 2^-57, to match the target in
+{{?TLS13}}. For AEAD_AES_128_GCM, the fourth term in this inequality dominates
+the rest, so the others can be removed without affecting the result. This
+produces the following approximation:
+
+~~~
+v <= 2^54
+~~~
+
+For AEAD_AES_256_GCM, the second and fourth terms dominate the rest, so the
+others can be removed without affecting the result. This produces the following
+approximation:
+
+~~~
+v <= 2^182
+~~~
+
+This is substantially smaller than the limit for AEAD_AES_128_GCM. Thus,
+for consistency, this document uses the larger of the two -- 2^54 -- for both.
```

Comment:
```suggestion
this document recommends that the smaller limit from AEAD_AES_128_GCM also
apply to AEAD_AES_256_GCM.
```

(The only important bit being s/larger/smaller/.)

This is partly consistency (because it is easier to use a single number here) and partly being conservative (because that large number is just unbelievable), and partly because that smaller value is already large enough.

Reply:
Woof. That one was bad. Good catch!